core/frontend/src/internal/tile/RealityModelTileTree.ts

```diff
@@ -665,12 +666,16 @@ export namespace RealityModelTileTree {
       }
       removals.push(context.viewport.onChangeView.addListener((vp, previousViewState) => {
         if(compareMapLayer(previousViewState, vp.view)){
-          const mapImagery = context.viewport.displayStyle.settings.mapImagery;
           this._layerRefHandler.setBaseLayerSettings(mapImagery.backgroundBase);
           this._layerRefHandler.setLayerSettings(mapImagery.backgroundLayers);
           this._layerRefHandler.clearLayers();
         }
       }));
+      removals.push(IModelApp.tileAdmin.onTileChildrenLoad.addListener(() => {

```

Comment:
My understanding is that this solves https://github.com/iTwin/itwinjs-core/issues/7818 by ensuring that the tile tree which is being draped is fully loaded before the draping begins.

I believe this code you added will trigger when any children tile loads, not just the entire tree.

You said elsewhere that onTileTreeLoad does not work; what did you observe exactly?